docs: Document password_file reload behavior for basic_auth (#6740)
grafana/alloy: docs/sources/reference/components/prometheus/prometheus.exporter.elasticsearch.md

{{< docs/shared lookup="reference/components/basic-auth-block.md" source="alloy" version="<ALLOY_VERSION>" >}}

{{< admonition type="note" >}}

> Unlike most components that use this block, `prometheus.exporter.elasticsearch` reads `password_file` once when the component starts or reloads its configuration, not on every outgoing request.

Reviewer: ```suggestion
`prometheus.exporter.elasticsearch` reads `password_file` once when it starts or reloads its configuration.
It doesn't read the file on every outgoing request.
```
